Vara external (#945)
se-passau/VaRA-Tool-Suite: docs/source/vara-ts/vara-external.rst

        Template example: https://github.com/se-sic/varats-oot-template/tree/test-oot
        Expected structure:
            /path/to/repo/projects
            /path/to/repo/experiments
            /path/to/repo/research_tools
            /path/to/repo/tables
            /path/to/repo/plots
            /path/to/repo/reports

Invalid Registered Repositories
...............................

If a previously registered repository becomes invalid (e.g., folders are deleted), **vara-external** will prompt you to confirm unregistering it:

.. code-block:: console

    The repository is already registered but does not match the template. Unregister it from the configuration? [y/N]:

- Type ``y`` to unregister the invalid repository
- Type ``n`` to keep it registered (for recovery/manual cleanup)

Already Registered Repositories
...............................

Attempting to register the same repository twice will skip registration:

.. code-block:: console

    Repository already registered at: /path/to/your/external-repository

Configuration
--------------

Registered external repositories are stored in the VaRA configuration file. The configuration key is:

> .. code-block:: ini

Reviewer: If sphinx supports YAML, that would be the actual language used in the config file.

```suggestion
.. code-block:: yaml
```

    external_source_repositories = [
        "/path/to/repo1",
        "/path/to/repo2"
    ]

> To manually modify registered repositories, edit your VaRA configuration file directly (usually at ``~/.vara/vara_config.yaml`` or equivalent on your system).

Reviewer: Our documentation uses `$VARATS_ROOT` as a system-agnostic shorthand for that directory.
Also, the config file name is incorrect here. 

```suggestion
To manually modify registered repositories, edit your VaRA configuration file directly (``$VARATS_ROOT/.varats.yaml``).
```

Example Workflow
----------------

> 

Reviewer: Maybe, as step 0 or prerequisite, refer to the "Getting Started" guide.

.. code-block:: console

    cd my-vara-extensions/
    # Add your projects, experiments, etc.

**Step 3: Register with VaRA-TS using vara-external**

Once your repository is ready, register it:

.. code-block:: console

    vara-external set /absolute/path/to/my-vara-extensions

**Step 4: Use in experiments**

Your custom projects and experiments are now available to VaRA-TS and can be used in experiment configurations.

Manual Setup (Alternative)
..........................

If you prefer to create the structure manually instead of cloning:

.. code-block:: console

    mkdir -p my-vara-extensions/{projects,experiments,research_tools,tables,plots,reports}

Template Repository
-------------------

For a complete example of a well-structured external repository with all required folders and structure, see the `VaRA Out-of-Tree Template <https://github.com/se-sic/varats-oot-template>`_.

**Recommended approach:**

1. **Clone** the template locally for a quick start
2. **Fork** the template on GitHub for a portable, version-controlled repository under your own account

> 3. Use `vara-external set` to register it with VaRA-TS

Reviewer: I think we can streamline this a bit by integrating these two alternatives directly into the **Step 1** part above.

> Troubleshooting

Reviewer: Not sure how necessary this is. All these commands are already mentioned above and the documentation for this tool is not that long or complicated.
